### ISSUE TITLE
"areaTypeAndComposition" should be fullfilled with more types

### ISSUE THREAD

**Issue by j3r3m1** [orbisgis/geoclimate#881]
Currently, when we make the GeoClimate calculations, only a small fraction of the buildings are considered in the TypeProportion process (https://github.com/orbisgis/geoclimate/blob/ffa10ae81e498a75f33789f20c14b64aa3dbf167/geoindicators/src/main/groovy/org/orbisgis/geoclimate/geoindicators/GenericIndicators.groovy#L677). The default values considered are given at the following line:
https://github.com/orbisgis/geoclimate/blob/ffa10ae81e498a75f33789f20c14b64aa3dbf167/geoindicators/src/main/groovy/org/orbisgis/geoclimate/geoindicators/WorkflowGeoIndicators.groovy#L372

We should definitely extend this map to at least consider the "house" keyword as a "house" fraction or include it in the "residential" type. We can add a test to check that the percentage of building taken into account is near 100%.

**Comment by j3r3m1** [comment]
@wurtzj

- which building types are used in TEB ? 
- would you be more interested by the building type or the building use (or both) ? Here are the possible type: https://github.com/orbisgis/geoclimate/wiki/Building-layer#type-and-main_use-column-values (we can gather them into main categories)
- do you want the fraction of each type based on the building area or building floor area ?

**Comment by j3r3m1** [comment]
OK, we still have to define what are the "main" types (@wurtzj).
The idea is then to directly create a new attribute at building input scale which will gather some of the existing types into a more generic one.

**Comment by wurtzj** [comment]
Hi @j3r3m1 @ebocher 



What is the difference between building area and building floor area ? walls ? number of levels ?
I would rather say we need the fraction of each use. So if there are commercial activities on first floor, and residential on other floors, then a fraction of each is ideal.

TEB deal with building usage with an approach is similar to a tile approach with flux
aggregation (e.g. using TEB for urban areas and ISBA for rural areas and aggregate fluxes towards
the atmosphere), but the tiles here represent different building uses and human behaviours and the
fluxes towards the building envelope (wall and roof) and the outdoor air are aggregated.

In TEB we use these types :

Fraction of individual housing
building use

Fraction of collective housing
building use

Fraction of commercial
building use

Fraction of tertiary building
use

Fraction of industrial building
use

Fraction of non heated
buildings

Fraction of low-rise building

Fraction of mid-rise building

Fraction of high-rise building

Fraction of activity buildings

Fraction of activity buildings

Fraction of households with
high Equipment-Intensity-of-
Use, collective housing

Fraction of households with
low Equipment-Intensity-of-
Use, individual housing

Fraction of households with
low Equipment-Intensity-of-
Use, collective housing

Fraction of households with
high Energy Control
Behaviour, individual housing

Fraction of households with
high Energy Control
Behaviour, collective housing


Robert Schoetter has already done some aggregation scripts to adapt the output of Geoclimate to TEB.
I can send it to you if you need.

**Comment by j3r3m1** [comment]
> So if there are commercial activities on first floor, and residential on other floors, then a fraction of each is ideal.
For now I am afraid we will not have such degree of detail. But at least I see that the right indicator is the fraction of building floor area for each type.

Thank you for sharing these informations. Even though you might have existing process to connect the GeoClimate output to these inputs, I try to see what would be the best using the current data we have as input
- [ ] Fraction of individual housing building use : sounds like fraction of "house" or "detached"
- [ ] Fraction of collective housing building use :  "apartments"
--> we will also have a residential fraction where we do not have more info about the type 

- [ ] Fraction of commercial building use : "commercial"
- [ ] Fraction of tertiary building use: "government", "townhall", "office"
- [ ] Fraction of industrial building use: "industrial"
- [ ] Fraction of non heated buildings: "silo" ?, "barn", "ruins"
- [ ] Fraction of low-rise building: the outputs from "roofFractionDistributionExact" should be used (histogram of building heights)
- [ ] Fraction of mid-rise building: the outputs from "roofFractionDistributionExact" should be used
- [ ] Fraction of high-rise building: the outputs from "roofFractionDistributionExact" should be used
- [ ] Fraction of activity buildings: I have no idea what this could be ?

I think the three following are based on a combination of the GeoClimate outputs + informations about the population. It would be good to know what they are anyway to be sure that the definition of our indicator is still the same.
- [ ] Fraction of households with high Equipment-Intensity-of- Use, collective housing
- [ ] Fraction of households with low Equipment-Intensity-of- Use, individual housing
- [ ] Fraction of households with low Equipment-Intensity-of- Use, collective housing
- [ ] Fraction of households with wigh Energy Control Behaviour, individual housing
- [ ] Fraction of households with high Energy Control Behaviour, collective housing

After that we still have quite a lot of information about building types which is not used. It is not a problem but maybe you would like some of them belong to a specific category above. You can have a look at the main use and type table and tell us:
https://github.com/orbisgis/geoclimate/wiki/Building-layer#type-and-main_use-column-values

**Comment by j3r3m1** [comment]
We need to update the type and main use with the "raw OSM data".
Once it is done, we need to update the GeoClimate doc to say:
- that types and main uses are directly the ones from OSM with their definition (removing other tables)
-  that aggregation of type are documented in the output indicators tables (LCZ and thermal energy indicators related)

**Comment by j3r3m1** [comment]
We should also check in the PR that the light industry tag has been managed for the LCZ (the initial combination of tags that have been aggregated to light_industry in the json should be performed using the typeProportion function specifically for the LCZ cases (+ for the building thermal params of course)

**Comment by j3r3m1** [comment]
> * Fraction of collective housing building use :  "apartments","barracks","abbey"
>       --> we will also have a residential fraction where we do not have more info about the type

@wurtzj By default if we have only the tag residential, do you want we put the building as individual or collective housing ?

**Comment by j3r3m1** [comment]
> * Fraction of tertiary building use: "government", "townhall", "retail","gambling","music_venue", "nightclub", "shop","store","supermarket","office","terminal","airport_terminal","bank","bureau_de_change","boat_rental","car_rental","research_institute","community_centre", "conference_centre","events_venue","exhibition_centre","social_centre","studio","theatre","library","healthcare",entertainment_arts_culture","hospital", "information"

@wurtzj do you remember a good reason to having the following types as tertiary ? I would rather go for commercial, is it a mistake or is it justified ?
"retail","gambling","music_venue", "nightclub", "shop","store","supermarket", "bank","bureau_de_change","boat_rental","car_rental"

**Comment by wurtzj** [comment]
Hi @j3r3m1 ,

Would it be possible to sort them using thresholds based on the building footprint and building height ? High and/or extensive buildings are more likely to be collective. 
For these types it is Ok the commercial. Any some of them cannot fit in any of these two categories.

**Comment by j3r3m1** [comment]
OK I will sort them in a specific category called "other_residential" and we will see how to combine this fraction to other informations outside GeoClimate

**Comment by j3r3m1** [comment]
The PR is now OK, the documentation has been updated (cf. links below) and a new GeoClimate version will come soon.
- https://github.com/orbisgis/geoclimate/wiki/Building-layer#type-and-main_use-column-values
- https://github.com/orbisgis/geoclimate/wiki/RSU-indicators#area_fraction_x
